pkgs/development/python-modules/grpcio/default.nix

```diff
@@ -1,36 +1,49 @@
 {
   lib,
   stdenv,
+  grpc,
   buildPythonPackage,
   fetchPypi,
-  grpc,
-  six,
-  protobuf,
-  enum34 ? null,
-  futures ? null,
-  isPy27,
-  pkg-config,
+
+  # build-system
+  setuptools,
+
+  # nativeBuildInputs
   cython,
+  pkg-config,
+
+  # buildInputs
   c-ares,
   openssl,
   zlib,
+
+  # dependencies
+  six,
+  protobuf,
+  isPy27,
+  enum34 ? null,
+  futures ? null,
 }:
 
 buildPythonPackage rec {
   pname = "grpcio";
-  format = "setuptools";
-  version = "1.64.1";
+  version = "1.66.1";
+  pyproject = true;
 
   src = fetchPypi {
     inherit pname version;
-    hash = "sha256-jVHdHFnV+g80JmuAo4BewpofJkJcKlRzYTP22H/Eloo=";
+    hash = "sha256-NTNPnJdFrdPjV+M3J1b9MtklvVLEHal/Tf2vveC/DuI=";
   };
 
   outputs = [
     "out"
     "dev"
   ];
 
+  build-dependencies = [
```

Comment:
I don't think this is a thing. `build-system` is:

```suggestion
  build-system = [
```

Without this change, I'm not able to build `.#python312Packages.grpcio`, it fails pretty quickly with an error about setuptools missing:

```
$ nix build .#python312Packages.grpcio
error: builder for '/nix/store/7kz6idym77sgv6xmvz7zf7fh2cl0dvmd-python3.12-grpcio-1.66.1.drv' failed with exit code 1;
       last 25 log lines:
       > Running phase: buildPhase
       > Executing pypaBuildPhase
       > Creating a wheel...
       > * Getting build dependencies for wheel...
       >
       > Traceback (most recent call last):
       >   File "/nix/store/mwvam3rmj6k3705z55griqzchlkjdajb-python3.12-pyproject-hooks-1.0.0/lib/python3.12/site-packages/pyproject_hooks/_impl.py", line 321, in _call_hook
       >     raise BackendUnavailable(data.get('traceback', ''))
       > pyproject_hooks._impl.BackendUnavailable: Traceback (most recent call last):
       >   File "/nix/store/mwvam3rmj6k3705z55griqzchlkjdajb-python3.12-pyproject-hooks-1.0.0/lib/python3.12/site-packages/pyproject_hooks/_in_process/_in_process.py", line 77, in _build_backend
       >     obj = import_module(mod_path)
       >           ^^^^^^^^^^^^^^^^^^^^^^^
       >   File "/nix/store/wfbjq35kxs6x83c3ncpfxdyl5gbhdx4h-python3-3.12.6/lib/python3.12/importlib/__init__.py", line 90, in import_module
       >     return _bootstrap._gcd_import(name[level:], package, level)
       >            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
       >   File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 1310, in _find_and_load_unlocked
       >   File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
       >   File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 1324, in _find_and_load_unlocked
       > ModuleNotFoundError: No module named 'setuptools'
       >
       > ERROR Backend 'setuptools.build_meta:__legacy__' is not available.
       For full logs, run 'nix log /nix/store/7kz6idym77sgv6xmvz7zf7fh2cl0dvmd-python3.12-grpcio-1.66.1.drv'.
```

Reply:
Definitely, it's an oversight. Fixed.